src/symmetric_crypto/nonce.rs

```diff
@@ -61,35 +56,21 @@ impl<const NONCE_LENGTH: usize> NonceTrait for Nonce<NONCE_LENGTH> {
         Ok(Self(b))
     }
 
-    fn increment(&self, increment: usize) -> Self {
-        let mut bi = BigUint::from_bytes_le(&self.0);
-        bi += BigUint::from(increment);
-        let mut bi_bytes = bi.to_bytes_le();
-        bi_bytes.resize(NONCE_LENGTH, 0);
-        Self(bi_bytes.try_into().expect("This should never happen"))
-    }
-
+    #[inline]
     fn xor(&self, b2: &[u8]) -> Self {
```

Comment:
The `b2` slice length is not checked? Maybe add an `assert!` or take a fixe size length?

Reply:
Yes indeed, I don't know why I changed to this code...